### PR TITLE
Refactor scores, stats and channels repos to build queries with sqlalchemy core 1.4

### PR DESCRIPTION
**Pull request by cmyui**
Co-authored-by: James Wilson <tsunyoku@users.noreply.github.com>

# Describe your changes

## Checklist
- [x] I've manually tested my code

### CODE REVIEW

File: app/repositories/scores.py [hunk 1]
```diff
@@ -1,47 +1,90 @@
...
+    __table_args__ = (
+        Index("scores_map_md5_index", map_md5),
+        Index("scores_score_index", score),
+        Index("scores_pp_index", pp),
+        Index("scores_mods_index", mods),
+        Index("scores_status_index", status),
+        Index("scores_mode_index", mode),
+        Index("scores_play_time_index", play_time),
+        Index("scores_userid_index", userid),
+        Index("scores_online_checksum_index", online_checksum),
```
Comment (cmyui) [inline app/repositories/scores.py:61]: ouu didn't notice, this should be a unique key at the ddl level

can make the change in another pr

**Review by tsunyoku — APPROVED** [review]
